tracing/tracers/otel/otel.go

```diff
@@ -0,0 +1,299 @@
+			var err error
+			config.insecure, err = strconv.ParseBool(val)
+			if err != nil {
+				return options{}, fmt.Errorf("failed to parse %s as bool: %w", val, err)
+			}
+		case "use-http":
+			var err error
+			config.useHTTP, err = strconv.ParseBool(val)
+			if err != nil {
+				return options{}, fmt.Errorf("failed to parse %s as bool: %w", val, err)
+			}
+		}
+	}
+
+	if config.ServiceName == "" {
+		config.ServiceName = defServiceName
+	}
+
+	if config.host == "" {
+		return options{}, fmt.Errorf("'otlp-exporter-endpoint' option not found, this option is mandatory")
+	}
+
+	config.tags = map[string]string{
+		lightstep.ComponentNameKey: config.ServiceName,
+	}
+
+	for k, v := range globalTags {
+		config.tags[k] = v
+	}
+
+	return config, nil
+}
+
+func (tb *tracerBuilder) setupGRPCExporter(ctx context.Context) error {
+	clientOptions := []otlptracegrpc.Option{
+		otlptracegrpc.WithHeaders(tb.opt.customHeaders),
+		otlptracegrpc.WithEndpoint(net.JoinHostPort(tb.opt.host, strconv.Itoa(tb.opt.port))),
+	}
+
+	if tb.opt.insecure {
+		clientOptions = append(clientOptions, otlptracegrpc.WithInsecure())
+	}
+
+	// OpenTelemetry only supports gzip or no compression:
+	// https://github.com/open-telemetry/opentelemetry-go/blob/exporters/otlp/otlptrace/v1.17.0/exporters/otlp/otlptrace/otlptracegrpc/options.go#L91
+	if tb.opt.compressionMethod == "gzip" {
+		clientOptions = append(clientOptions, otlptracegrpc.WithCompressor(tb.opt.compressionMethod))
+	}
+
+	client := otlptracegrpc.NewClient(clientOptions...)
+
+	e, err := otlptrace.New(ctx, client)
+	if err != nil {
+		return err
+	}
+
+	tb.exporter = e
+	return nil
+}
+
+func (tb *tracerBuilder) setupHTTPExporter(ctx context.Context) error {
+	clientOptions := []otlptracehttp.Option{
+		otlptracehttp.WithHeaders(tb.opt.customHeaders),
+		otlptracehttp.WithURLPath(tb.opt.customPath),
+		otlptracehttp.WithEndpoint(net.JoinHostPort(tb.opt.host, strconv.Itoa(tb.opt.port))),
+	}
+
+	if tb.opt.insecure {
+		clientOptions = append(clientOptions, otlptracehttp.WithInsecure())
+	}
+
+	// OpenTelemetry only supports gzip or no compression:
+	// https://github.com/open-telemetry/opentelemetry-go/blob/3c476ce1816ae6f38758e90cc36d8b77ebcc223b/exporters/otlp/otlptrace/internal/otlpconfig/optiontypes.go#L29
+	if tb.opt.compressionMethod == "gzip" {
+		clientOptions = append(clientOptions, otlptracehttp.WithCompression(otlptracehttp.GzipCompression))
+	}
+
+	client := otlptracehttp.NewClient(clientOptions...)
+
+	e, err := otlptrace.New(ctx, client)
+	if err != nil {
+		return err
+	}
+
+	tb.exporter = e
+	return nil
+}
+
+func (tb *tracerBuilder) Build(ctx context.Context) (trace.Tracer, error) {
+	var err error
+	if tb.opt.useHTTP {
+		err = tb.setupHTTPExporter(ctx)
+	} else {
+		err = tb.setupGRPCExporter(ctx)
+	}
+	if err != nil {
+		return nil, err
+	}
+
+	tags := []attribute.KeyValue{
+		semconv.ServiceNameKey.String(tb.opt.ServiceName),
+		semconv.ServiceVersionKey.String(tb.opt.ServiceVersion), // ?
```

Comment:
what's the question?
Sounds good to have the version mandatory if you ask me

Reply:
Oh, that was a comment I forgot to delete.